### ISSUE TITLE
Adding custom ObservationHandler does not preserve the default ObservationHandlers

### ISSUE THREAD

**Issue by violetagg** [spring-projects/spring-boot#34510]
### Spring Boot version
`3.0.4`

### Reproducible example
https://github.com/violetagg/netty-observation-repro

### Problem description

I would like to use Reactor Netty tracing capabilities in Spring Boot application.
For this purpose I need to reuse the `ObservationRegistry` created by Spring Boot and to add `ObservationHandler`s provided by Reactor Netty.

Observations:
1. If I only reuse the `ObservationRegistry` without adding the `ObservationHandler`s provided by Reactor Netty - the tracing information is correctly visualised but the span tags are duplicated and one and the same Timer object is created every time.

![Screenshot 2023-03-08 at 9 42 13](https://user-images.githubusercontent.com/696661/223651704-ea601291-fba1-4e6f-8702-90f3ac4d7f7a.png) ![Screenshot 2023-03-08 at 9 42 25](https://user-images.githubusercontent.com/696661/223651716-9eada160-e4f6-48cb-af21-62359db63726.png)

2.  If I reuse the `ObservationRegistry` and add `ObservationHandler`s provided by Reactor Netty - the tracing information is **NOT** correctly visualised.

3. If I reuse the `ObservationRegistry`, add `ObservationHandler`s provided by Reactor Netty and in addition add all default `ObservationHandler`s provided by Spring Boot - the tracing information is correctly visualised, the span tags are not duplicated and the Timer object is not created always. Reactor Netty tracing capabilities are working as expected.

![Screenshot 2023-03-08 at 9 51 18](https://user-images.githubusercontent.com/696661/223653653-e54097d3-3bcc-4d20-a588-e4d29776a42f.png) ![Screenshot 2023-03-08 at 9 51 29](https://user-images.githubusercontent.com/696661/223653664-38388ed5-4888-4c31-bc63-2d969303df0d.png)

### Desired solution
Spring Boot should preserve the default `ObservationHandler`s when custom `ObservationHandler`s are added to the configuration.

**Comment by violetagg** [comment]
yep, because of the mentioned issue, in the example I added `@Order` when declaring the various `ObservationHandler`s

**Comment by marcingrzejszczak** [comment]
I think that the defaults observation handlers should not have conditional on missing bean

**Comment by marcingrzejszczak** [comment]
If you don't want the default tracing handler you register your own tracing handler before the default one because only one can be used. Same for metrics. We are grouping all handlers for exactly this reason - that you can put your own handlers before the existing ones.

Another option for @violetagg would be never to extend from the ones that we provide but delegate to them (that way the types would not be the same and `@ConditionalOnMissingBean` wouldn't have effect).

**Comment by violetagg** [comment]
> If you don't want the default tracing handler you register your own tracing handler before the default one because only one can be used. Same for metrics. We are grouping all handlers for exactly this reason - that you can put your own handlers before the existing ones.

Yes we do exactly that our tracing handlers are implemented in a way that `io.micrometer.tracing.handler.TracingObservationHandler#supportsContext` handles only our Reactor Netty implementation, everything else should be handled by the default configuration in Spring Boot.

The scenario here is that we do want the default Spring Boot tracing for the server. What we want to enable is the tracing for Reactor Netty HttpClient only.

> 
> Another option for @violetagg would be never to extend from the ones that we provide but delegate to them (that way the types would not be the same and `@ConditionalOnMissingBean` wouldn't have effect).

**Comment by marcingrzejszczak** [comment]
> The scenario here is that we do want the default Spring Boot tracing for the server. What we want to enable is the tracing for Reactor Netty HttpClient only.

Yeah so all of your custom handlers wouldn't extend from the default ones, they would delegate to them. So you would never hit the `@ConditionalOnMissingBean` problem

**Comment by violetagg** [comment]
> > The scenario here is that we do want the default Spring Boot tracing for the server. What we want to enable is the tracing for Reactor Netty HttpClient only.
> 
> Yeah so all of your custom handlers wouldn't extend from the default ones, they would delegate to them. So you would never hit the `@ConditionalOnMissingBean` problem

So we have to delegate to the default in both cases when in Spring Boot application and when in standalone Reactor Netty.
For Spring Boot we will delegate to the handlers created by Spring Boot.
In the standalone case to what should we delegate?

**Comment by violetagg** [comment]
@marcingrzejszczak and I tried the variant with delegation. (https://github.com/reactor/reactor-netty/commit/e02e72257fe0d801357bc1518f52cab521aab7ef)
Although technically this is possible it requires 3 `TracingObservationHandler`s which do not have functional logic but just delegate and this should be a way of implementing custom `TracingObservationHandler` and registering it as a `Bean` for every library not just Reactor Netty.

On the other hand if default `TracingObservationHandler`s are not conditional and we rely on the `io.micrometer.tracing.handler.TracingObservationHandler#supportsContext` then if the custom `TracingObservationHandler` supports this context it will be the one that will be chosen otherwise the default `TracingObservationHandler`s. (as `FirstMatchingCompositeObservationHandler` is used when configuring the `ObservationRegistry`)
